src/Engine/ProtoCore/Utils/ParserUtils.cs

```diff
@@ -82,6 +83,42 @@ public static IEnumerable<Node> Collect(CodeBlockNode node)
             }
         }
 
+        public class AssemblyDetector : AstTraversal
+        {
+            private ClassTable classTable;
+
+            private AssemblyDetector(ClassTable classTable)
+            {
+                this.classTable = classTable;
+            }
+
+            public static void GetAssemblyFromCodeBlock(ClassTable classTable, IEnumerable<Node> astNodes)
+            {
+                var assemblyDetector = new AssemblyDetector(classTable);
+                var assocNodes = astNodes.OfType<AssociativeNode>();
+                //assocNodes.Select(astNode => astNode.Accept(assemblyDetector));
+                assocNodes.ElementAt(0).Accept(assemblyDetector);
+            }
+
+            public override bool VisitIdentifierListNode(IdentifierListNode node)
+            {
+                var className = node.LeftNode.ToString();
+                var assembly = CoreUtils.GetAssemblyFromClassName(classTable, className);
```

Comment:
This utility method retrieves the assembly name from the classname used in the CBN script.